File: backend/app/schemas/analytics.py
from datetime import datetime

from pydantic import BaseModel

from app.models.document import Topic


class TopicCount(BaseModel):
    topic: Topic
    count: int


class LatencyStats(BaseModel):
    avg_ms: float | None
    p50_ms: float | None
    p95_ms: float | None


class FeedbackBreakdown(BaseModel):
    helpful: int
    not_helpful: int
    ratio_helpful: float | None  # None when total == 0, not a divide-by-zero 0.0


class AnalyticsSummary(BaseModel):
    since: datetime | None
    total_conversations: int
    total_turns: int
    grounded_rate: float | None
    clarification_rate: float | None
    topic_distribution: list[TopicCount]
    feedback: FeedbackBreakdown
    latency: LatencyStats
    corpus_document_count: int

File: frontend/src/types/analytics.ts
export interface TopicCount {
  topic: string;
  count: number;
}

export interface LatencyStats {
  avg_ms: number | null;
  p50_ms: number | null;
  p95_ms: number | null;
}

export interface FeedbackBreakdown {
  helpful: number;
  not_helpful: number;
  ratio_helpful: number | null;
}

export interface AnalyticsSummary {
  since: string | null;
  total_conversations: number;
  total_turns: number;
  grounded_rate: number | null;
  clarification_rate: number | null;
  topic_distribution: TopicCount[];
  feedback: FeedbackBreakdown;
  latency: LatencyStats;
  corpus_document_count: number;
}
